website/src/components/broadcast/roots/TallyTransmitter.vue

```diff
@@ -0,0 +1,47 @@
+<script>
+export default {
+    name: "TallyTransmitter",
+    props: ["client"],
+    data: () => ({
+        active: false,
+        visible: false,
+        scene: ""
+    }),
+    computed: {
+        state() {
+            if (this.active && this.visible) {
+                return "active";
+            } else if (this.visible) {
+                return "preview";
+            } else {
+                return "inactive";
+            }
+        }
+    },
+
+    methods: {
+        transmitState() {
+            this.$socket.client.emit("tally_change", { clientName: this.client.key, state: this.state, sceneName: this.scene });
+        }
+    },
+    mounted() {
+        window.addEventListener("obsSourceActiveChanged", (e) => {
+            if (!this.client) return;
+            window.obsstudio?.getCurrentScene((scene) => {
+                this.scene = scene?.name;
+            });
+            this.active = e.detail.active;
+            this.transmitState();
+        });
+
+        window.addEventListener("obsSourceVisibleChanged", (e) => {
+            if (!this.client) return;
+            window.obsstudio?.getCurrentScene((scene) => {
+                this.scene = scene?.name;
+            });
+            this.visible = e.detail.visible;
+            this.transmitState();
+        });
+    }
```

Comment:
I think this should do it?
```suggestion
    methods: {
        transmit() {
            this.$socket.client.emit("tally_change", { clientName: this.client.key, state: this.state, sceneName: this.scene });
        },
        transmitState() {
            if (window.obsstudio) {
                window.obsstudio.getCurrentScene((scene) => {
                    this.scene = scene?.name;
                    this.transmit();
                });
            } else {
                this.transmit();
            }
        }
    },
    mounted() {
        window.addEventListener("obsSourceActiveChanged", (e) => {
            if (!this.client) return;
            this.active = e.detail.active;
            this.transmitState();
        });

        window.addEventListener("obsSourceVisibleChanged", (e) => {
            if (!this.client) return;
            this.visible = e.detail.visible;
            this.transmitState();
        });
    }
```

Reply:
I wish the diff was a bit more helpful - basically moving the function calls into one method so it can check for the OBS object, and either wait for the scene name or just transmit.